packages/@react-spectrum/list/src/ListView.tsx

```diff
@@ -185,6 +200,67 @@ function ListView<T extends object>(props: ListViewProps<T>, ref: DOMRef<HTMLDiv
     });
   }
 
+  let dropState: DroppableCollectionState;
+  let droppableCollection: DroppableCollectionResult;
+  let isRootDropTarget: boolean;
+  if (isListDroppable) {
+    dropState = dropHooks.useDroppableCollectionState({
+      collection: state.collection,
+      selectionManager: state.selectionManager
+    });
+    droppableCollection = dropHooks.useDroppableCollection({
+      keyboardDelegate,
+      getDropTargetFromPoint(x, y) {
+        let closest = null;
+        let closestDistance = Infinity;
+        let closestDir = null;
+
+        x += domRef.current.scrollLeft;
+        y += domRef.current.scrollTop;
+
+        let visible = layout.getVisibleLayoutInfos(new Rect(x - 50, y - 50, x + 50, y + 50));
+
+        for (let layoutInfo of visible) {
+          let r = layoutInfo.rect;
+          let points: [number, number, string][] = [
+            [r.x, r.y + 4, 'before'],
+            [r.maxX, r.y + 4, 'before'],
+            [r.x, r.maxY - 8, 'after'],
+            [r.maxX, r.maxY - 8, 'after']
+          ];
+
+          for (let [px, py, dir] of points) {
+            let dx = px - x;
+            let dy = py - y;
+            let d = dx * dx + dy * dy;
+            if (d < closestDistance) {
+              closestDistance = d;
+              closest = layoutInfo;
+              closestDir = dir;
+            }
+          }
+
+          // TODO: Best way to implement only for when closest can be dropped on
+          if (y >= r.y + 10 && y <= r.maxY - 10 && state.collection.getItem(closest.key).value.type === 'folder') {
+            closestDir = 'on';
+          }
```

Comment:
Any thoughts on this? Is there a better way / is this too opinionated

Reply:
Without spending a bunch of time thinking about this, I cannot figure out what is happening here, can you write a comment explaining it? then I may have some opinions

Reply:
The base cases are dropping "before" or "after" and item. This is the case for dropping "on" an item. But we probably don't want all items to be able to be dropped on, so do we only factor in dropping "on" as a possibility for items with type="folder"? (This is the current behavior)

Reply:
IMO possibly too opinionated. Is there a way for a user to pass a function that returns if an item is droppable or not (similar to `allowsDraggingItem`)?

Reply:
maybe we do folder for now and open it later?